### PR TITLE
add support for OAI e2e tests for fedora34

### PR DESCRIPTION
**Pull request by vjayaramrh**
**What type of PR is this?**
> Uncomment only one ` /kind <>` line, hit enter to put that in a new line, and remove leading whitespaces from that line:
>
> /kind bug
> /kind cleanup
> /kind design
> /kind documentation
> /kind failing-test
> /kind feature
> /kind flake

/kind feature

**What this PR does / why we need it**:
This PR adds support or running OAI e2e tests on the Fedora 34 OS

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

**Comment by vjayaramrh** [comment]
FYI retesting with @arora-sagar's updated mysql image version of 8.0 (https://github.com/OPENAIRINTERFACE/oai-packages/blob/r2/database/mysql.yaml#L105). The earlier version of mysql:5.7 container image  was causing failures.

**Comment by tliron** [comment]
I'm sorry if this is obvious, but: Why Fedora 34, an OS from the year 2020? Why not target the latest?

**Comment by vjayaramrh** [comment]
> I'm sorry if this is obvious, but: Why Fedora 34, an OS from the year 2020? Why not target the latest?

free5gc gtpg module works only on a kernel version 5.15.x which is in Fedora 34, hence piggybacking on that effort for the time being.

**Comment by arora-sagar** [comment]
@vjayaramrh Is it possible to have a different flavor for OAI test scenario? I think in Ubuntu also we have different flavors 20 (Free5GC) and 22 (OAI).

**Comment by vjayaramrh** [comment]
> @vjayaramrh Is it possible to have a different flavor for OAI test scenario? I think in Ubuntu also we have different flavors 20 (Free5GC) and 22 (OAI).

@arora-sagar I agree with you, I want to first go to the extent possible with Fedora34 and document issues faced, if any. I next plan to go to a later version of Fedora.

**Comment by vjayaramrh** [comment]
@arora-sagar FYI This evening I was able to succesfully run the OAI e2e tests manually on Fedora 34 (created in GCP). The next step is to figure out why its not working as part of prow job invocation.

```
[fedora@nephio-r2-e2e-vish-fedora-3 ~]$ UE_POD=$(kubectl get pods -n oai-ue --context edge-admin@edge  -l app.kubernetes.io/name=oai-nr-ue -o jsonpath='{.items[*].metadata.name}')
UPF_POD=$(kubectl get pods -n oai-core --context=edge-admin@edge -l workload.nephio.org/oai=upf -o jsonpath='{.items[*].metadata.name}')
UPF_tun0_IP_ADDR=$(kubectl exec -it $UPF_POD -n oai-core -c upf-edge --context edge-admin@edge -- ip -f inet addr show tun0 | sed -En -e 's/.*inet ([0-9.]+).*/\1/p')
kubectl exec -it $UE_POD -n oai-ue --context edge-admin@edge -- ping -I oaitun_ue1 -c 3 $UPF_tun0_IP_ADDR
PING 10.0.0.1 (10.0.0.1) from 10.0.0.2 oaitun_ue1: 56(84) bytes of data.
64 bytes from 10.0.0.1: icmp_seq=1 ttl=64 time=22.0 ms
64 bytes from 10.0.0.1: icmp_seq=2 ttl=64 time=12.3 ms
64 bytes from 10.0.0.1: icmp_seq=3 ttl=64 time=15.5 ms

--- 10.0.0.1 ping statistics ---
3 packets transmitted, 3 received, 0% packet loss, time 2002ms
rtt min/avg/max/mdev = 12.306/16.624/22.041/4.049 ms

```

File: .prow.yaml [hunk 1]
```diff
@@ -246,6 +246,49 @@ presubmits:
                 path: id_rsa
               - key: id_rsa.pub
                 path: id_rsa.pub
+  - name: e2e-oai-fedora-34
+    annotations:
+    labels:
+    run_if_changed: '^e2e/|.prow.yaml'
```
Comment (electrocucaracha) [inline .prow.yaml:252]: This changes won't be necessary in the final version
```suggestion
    run_if_changed: '^e2e/'
```

**Review by electrocucaracha — CHANGES_REQUESTED** [review]
@vjayaramrh in order to merge this, could you provide information in the description section?

**Comment by electrocucaracha** [comment]
> > I'm sorry if this is obvious, but: Why Fedora 34, an OS from the year 2020? Why not target the latest?
> 
> free5gc gtpg module works only on a kernel version 5.15.x which is in Fedora 34, hence piggybacking on that effort for the time being. In the future, we have to investigate moving to a later version and identify gaps if and why we cannot move to a later version.

[The gtpg module won't be loaded on OAI deployments](https://github.com/nephio-project/test-infra/blob/main/e2e/provision/playbooks/roles/bootstrap/tasks/main.yml#L16) so maybe we can start our testing on latest Fedora releases.

**Comment by electrocucaracha** [comment]
@vjayaramrh don't forget to include this config definition in the periodic task section, like this one https://github.com/nephio-project/test-infra/blob/main/prow/config/config.yaml#L1551-L1596

**Comment by vjayaramrh** [comment]
> @vjayaramrh don't forget to include this config definition in the periodic task section, like this one https://github.com/nephio-project/test-infra/blob/main/prow/config/config.yaml#L1551-L1596

@electrocucaracha thanks, how about I open a github issue "add periodic task for fedora34 OAI daily test run" so that it can be a low hanging fruit for contribution by someone new to test-infra project? Let me know

**Review by electrocucaracha — APPROVED** [review]
/lgtm
/approve

**Pull request merged**